backend/manager/modules/vdsbroker/src/main/java/org/ovirt/engine/core/vdsbroker/monitoring/VmAnalyzer.java

```diff
@@ -746,7 +746,9 @@ private void updateRuntimeData() {
             vmGuestAgentNics = filterGuestAgentInterfaces(nullToEmptyList(vdsmVm.getVmGuestAgentInterfaces()));
             dbVm.setIp(extractVmIps(vmGuestAgentNics));
         }
-
+        if (!Objects.equals(vdsmVm.getVmDynamic().getAppList(), dbVm.getAppList())) {
+            resourceManager.refreshIsoCache(vdsManager.getCopyVds().getStoragePoolId());
```

Comment:
Do we need to call getCopyVds() just to get the storage pool id?

Reply:
Yes, AFAIK. There is no other place where the storage pool ID is stored.